### ISSUE TITLE
Can not get ssid when switch from mobile data to wifi on Android

### ISSUE THREAD

**Issue by lisongmin** [SagerNet/sing-box#1414]
### Operating system

Android

### System version

lienageos 20

### Installation type

sing-box for Android Graphical Client

### If you are using a graphical client, please provide the version of the client.

1.8.4

### Description

I config a wireguard outbound to access home network from outside, there are two route rules:

1. It should connect directly if i'm at home
2. It should connect via wireguard outbound otherwise

```json
      {
        "wifi_ssid": ["test"],
        "ip_cidr": ["192.168.2.0/24"],
        "outbound": "direct"
      },
      {
        "ip_cidr": ["192.168.2.0/24"],
        "outbound": "go-home"
      }
```

I can access home network when i tape the start button on the sing-box client.
But after that, if i switch from mobile to wifi, I can't access home network any more.
here's some log when I switch to wifi:

```
INFO router: updated default interface wlan0, index 23
DEBUG outbound/wireguard[go-home]: routine: receive incoming receive - stopped
INFO router: updated WIFI state: SSID=<unknown ssid>, BSSID=02:00:00:00:00:00
```

It seems we need the ACCESS_BACKGROUND_LOCATION permission to get wifi at the background. 

The configration and logs:

* [sfa.json](https://github.com/SagerNet/sing-box/files/14076118/sfa.json)
* [sfa.log](https://github.com/SagerNet/sing-box/files/14076103/sfa.log)

### Reproduction

1. Disable wifi
2. Start sing-box
3. Access home network like `curl http://192.168.2.1` and it works
4. Enable wifi
5. Try to access home network again, but failed now. 

### Integrity requirements

- [X] I confirm that I have read the documentation, understand the meaning of all the configuration items I wrote, and did not pile up seemingly useful options or default values.
- [X] I confirm that I have provided the server and client configuration files and process that can be reproduced locally, instead of a complicated client configuration file that has been stripped of sensitive data.
- [X] I confirm that I have provided the simplest configuration that can be used to reproduce the error I reported, instead of depending on remote servers, TUN, graphical interface clients, or other closed-source software.
- [X] I confirm that I have provided the complete configuration files and logs, rather than just providing parts I think are useful out of confidence in my own intelligence.

**Comment by nekohasekai** [comment]
Try 1.8.5

**Comment by lisongmin** [comment]
This problem is fixed in 1.8.5. 

thanks your great work.
